packages/examples/core/ts/platform/platform.ts

```diff
@@ -6,14 +6,36 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Component, createPlatformFactory} from '@angular/core';
-import {platformBrowserDynamic} from '@angular/platform-browser-dynamic';
+// #docregion componentSelector
+@NgModule({imports: [BrowserModule], declarations: [ComponentOne, ComponentTwo]})
+export class AppModule implements DoBootstrap {
+  readonly componentMap = {'ComponentOne': ComponentOne, 'ComponentTwo': ComponentTwo};
 
-// #docregion longform
-@Component({selector: 'my-app', template: 'Hello World'})
-class MyApp {
+  ngDoBootstrap(app: ApplicationRef) {
+    this.fetchDataFromApi().then((componentName) => {
+      app.bootstrap(this.componentMap[componentName]);
+    });
+  }
+
+  fetchDataFromApi(): Promise<string> {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve('ComponentTwo');
+      }, 2000);
+    });
+  }
```

Comment:
I think the implementation of method does not add value to the understanding, so it could be hidden in the code snippet that appears in the docs by adding further docregion markers:

```suggestion
// #enddocregion componentSelector
  fetchDataFromApi(): Promise<string> {
    return new Promise((resolve) => {
      setTimeout(() => {
        resolve('ComponentTwo');
      }, 2000);
    });
  }
// #docregion componentSelector
```